hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -485,9 +497,14 @@ def sampled_from(elements):
 @cacheable
 @defines_strategy
 def lists(
-    elements=None, min_size=None, average_size=None, max_size=None,
-    unique_by=None, unique=False,
+    elements=None,  # type: SearchStrategy
+    min_size=None,  # type: int
+    average_size=None,  # type: int
+    max_size=None,  # type: int
+    unique_by=None,  # type: Callable
```

Comment:
What does `type: Callable` mean? `Callable` is a generic type, so it should e.g. by `Callable[Any, Any]`. I can't find any documentation to the effect of what generic types in mypy are supposed to represent when used without their parameters.

Reply:
In this case it's equivalent to `Callable[..., Any]` or to plain `callable`.  I thought it clear as the bare generic type but can switch if you prefer.

We *could* specify `Callable[[Any], Any]` (note: `[]` around args), but would lead to type-checking errors for functions with optional arguments - the idea of keyword and optional arguments is a Mypy extension only.  I did this for the argument to `deferred()`, but thought that a more flexible approach was justified here.